### PR TITLE
Bug 734469 - Clients stage table missing during sync.

### PR DESCRIPTION
**Pull request by emtwo**
Main issue was static cached references to readableDatabase & writableDatabase. Since the history dataExtender did not close() after it was done its work, the clients database was trying to use the cached version of the history database which, of course, didn't have a clients table (eeeekkk!!)

That was such a fun bug :D I love how it threw me off so much :D

### CODE REVIEW

**Comment by emtwo** [comment]
You should probably add a test

**Comment by ncalexan** [comment]
Okay, defer until rnewman reviews.

**Comment by emtwo** [comment]
Added tests & the such.

**Comment by emtwo** [comment]
- Closing dataExtender upon session.finish() & session.abort() now
- Added appropriate tests for that

**Comment by ncalexan** [comment]
+1 from me.

File: test/src/org/mozilla/android/sync/test/TestCachedSQLiteOpenHelper.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
+package org.mozilla.android.sync.test;
```
Comment (rnewman) [inline test/src/org/mozilla/android/sync/test/TestCachedSQLiteOpenHelper.java:4]: License blocks.

File: test/src/org/mozilla/android/sync/test/TestCachedSQLiteOpenHelper.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    clientsDB = new ClientsDatabase(mContext);
+    extender = new AndroidBrowserHistoryDataExtender(mContext);
+  }
+
+  public void tearDown() {
+    clientsDB.close();
+    extender.close();
+  }
+
+  public void testUnclosedDatabasesDontInteact() throws NullCursorException {
```
Comment (rnewman) [inline test/src/org/mozilla/android/sync/test/TestCachedSQLiteOpenHelper.java]: s/Inteact/Interact

**Comment by rnewman** [comment]
Looks good to me. Fix those two nits and merge!

**Pull request merged**